scrapy_poet/api.py

```diff
@@ -104,29 +105,27 @@ def parse(self, response):
     disk queues, because in this case Scrapy is able to serialize
     your request object.
     """
-    if not issubclass(page_cls, ItemPage):
-        raise TypeError(f"{page_cls.__name__} should be a subclass of ItemPage.")
-
-    if getattr(page_cls.to_item, "__isabstractmethod__", False):
-        raise NotImplementedError(
-            f"{page_cls.__name__} should implement to_item method."
-        )
-
-    # When the callback is used as an instance method of the spider, it expects
-    # to receive 'self' as its first argument. When used as a simple inline
-    # function, it expects to receive a response as its first argument.
-    #
-    # To avoid a TypeError, we need to receive a list of unnamed arguments and
-    # a dict of named arguments after our injectable.
-    def parse(*args, page: page_cls, **kwargs):  # type: ignore
-        yield page.to_item()  # type: ignore
-
-    async def async_parse(*args, page: page_cls, **kwargs):  # type: ignore
-        yield await page.to_item()  # type: ignore
-
-    if iscoroutinefunction(page_cls.to_item):
-        setattr(async_parse, _CALLBACK_FOR_MARKER, True)
-        return async_parse
+    if issubclass(page_or_item_cls, ItemPage):
+        # When the callback is used as an instance method of the spider, it expects
+        # to receive 'self' as its first argument. When used as a simple inline
+        # function, it expects to receive a response as its first argument.
+        #
+        # To avoid a TypeError, we need to receive a list of unnamed arguments and
+        # a dict of named arguments after our injectable.
```

Comment:
this comment applies to both code paths; it seem to explain why *args and **kwargs are needed